kong/plugins/rate-limiting/policies/init.lua

```diff
@@ -99,6 +99,7 @@ return {
         end
       end
 
+	  local t = {}
```

Comment:
Could we come up with a better name for this variable than `t`? Maybe `keys`?

I would also suggest reducing the GC pressure by creating two tables and an index variable here:
```lua
local keys = {}
local expirations = {}
local idx = 0
```

Then, the logic in the `pairs(periods)` loop becomes:
```lua
idx = idx + 1
keys[idx] = cache_key

if not exists or exists == 0 then
  expirations[idx] = EXPIRATIONS[period]
end 
```

Doing this will:
- Only create 2 tables instead of creating a table per limit
- Avoid the O(n) operation implied by the length operator `#` (it is a typical LuaJIT idiom to keep track of indexes separately as we are building lists)
- Reduce branching

Then, the pipelining logic becomes:
```lua
for i = 1, idx do
  red:incrby(keys[i], value)
  if expirations[i] then
    red:expire(keys[i], expirations[i])
  end
end
```

Once again we avoid the `#` operator.

Reply:
style: this indentation is also completely off, please respect the 2 spaces indentation encountered across this codebase.